tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/HotChocolate/HotChocolateCommon.cs

```diff
@@ -114,12 +115,14 @@ internal static void RecordExecutionErrorsIfPresent(Span span, string errorType,
 
             if (errorCount > 0)
             {
-                RecordExecutionErrors(span, errorType, errorCount, ConstructErrorMessage(executionErrors));
+                RecordExecutionErrors(span, errorType, errorCount, ConstructErrorMessageAndEvents(executionErrors, out var errorSpanEvents), errorSpanEvents);
             }
         }
 
-        private static string ConstructErrorMessage(List<IError> executionErrors)
+        private static string ConstructErrorMessageAndEvents(List<IError> executionErrors, out List<SpanEvent> spanEvents)
```

Comment:
nitpicky but I think this would be better off as two functions, one to `ConstructErrorMessage` another `ConstructErrorSpanEvents`
I think that separation of concerns will make it easier to maintain/read in the future.

This would mean then that we need to enumerate `executionErrors` twice, but I'm _assuming_ that the performance hit there would be negligible

Reply:
+1 - `ConstructErrorMessageAndEvents` is essentially returning two different values so I think it will be more maintainable to break this out into two separate functions where each one returns one of the values

Reply:
Updated in the latest commits along with other nits and recommendations.